src/Controller/CRUDController.php

```diff
@@ -66,11 +68,29 @@ class CRUDController extends Controller
      */
     private $templateRegistry;
 
-    public function setContainer(?ContainerInterface $container = null): void
+    public function setContainer(ContainerInterface $container): ?ContainerInterface
     {
-        $this->container = $container;
+        $result = parent::setContainer($container);
 
         $this->configure();
+
+        return $result;
+    }
+
+    public static function getSubscribedServices()
+    {
+        return parent::getSubscribedServices() + [
+            'sonata.admin.pool' => Pool::class,
+            'sonata.admin.breadcrumbs_builder' => BreadcrumbsBuilderInterface::class,
+            'sonata.admin.audit.manager' => AuditManagerInterface::class,
+            'sonata.admin.object.manipulator.acl.admin' => AdminObjectAclManipulator::class,
+            'sonata.exporter.exporter' => '?'.Exporter::class,
+            'sonata.admin.admin_exporter' => '?'.AdminExporter::class,
+            'sonata.admin.security.acl_user_manager' => '?'.AdminAclUserManagerInterface::class,
+
+            'logger' => '?'.LoggerInterface::class,
+            'translator' => TranslatorInterface::class,
+        ];
```

Comment:
```suggestion
        return [
            'sonata.admin.pool' => Pool::class,
            'sonata.admin.breadcrumbs_builder' => BreadcrumbsBuilderInterface::class,
            'sonata.admin.audit.manager' => AuditManagerInterface::class,
            'sonata.admin.object.manipulator.acl.admin' => AdminObjectAclManipulator::class,
            'sonata.exporter.exporter' => '?'.Exporter::class,
            'sonata.admin.admin_exporter' => '?'.AdminExporter::class,
            'sonata.admin.security.acl_user_manager' => '?'.AdminAclUserManagerInterface::class,

            'logger' => '?'.LoggerInterface::class,
            'translator' => TranslatorInterface::class,
        ] + parent::getSubscribedServices();
```

Reply:
Are you sure about this change? For now it is not the case, but we are adding new services to the previous one, not the other way around (so in case we want to override some of the parent services, your change won't work for that matter)

Reply:
I'd not say it won't work.
I think keeping the order of precedence is important in order to preserve the relevance of the services that we are choosing by our side, in case that the parent method makes changes, adding some of these services.

Reply:
I don't know if you could saw this previous comment in this matter https://github.com/sonata-project/SonataAdminBundle/pull/6476#discussion_r509091257.

Reply:
Sorry, didnt saw your previous comment. Applied (I thought the order for + and array merge was the same